src/main/java/org/broadinstitute/hellbender/utils/variant/writers/ShardingVCFWriter.java

```diff
@@ -0,0 +1,161 @@
+package org.broadinstitute.hellbender.utils.variant.writers;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.util.FileExtensions;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.variantcontext.writer.Options;
+import htsjdk.variant.variantcontext.writer.VariantContextWriter;
+import htsjdk.variant.vcf.VCFHeader;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+import org.broadinstitute.hellbender.utils.variant.GATKVariantContextUtils;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Variant writer tha splits output to multiple VCFs given the maximum records per file. Before using {@link #add},
+ * the header should be set using either {@link #setHeader} or {@link #writeHeader}, which may only be called once
+ * and will determine whether headers are written to all shards.
+ *
+ * @author Mark Walker &lt;markw@broadinstitute.org&gt;
+ */
+public class ShardingVCFWriter implements VariantContextWriter {
```

Comment:
Is this going to index as well?

Reply:
Yes, that is specified through the options parameter